volume/drivers/nfs/nfs.go

```diff
@@ -211,14 +212,18 @@ func (d *driver) Create(
 	source *api.Source,
 	spec *api.VolumeSpec) (string, error) {
 
-	volumeID := locator.Name
-	if volumeID == "" && source.Parent != "" {
-		volumeID = d.getNewSnapVolID(source.Parent)
-		dlog.Infof("Creating snap vol id: %s", volumeID)
+	if locator.Name == "" {
+		return "", errors.New("volume name cannot be empty")
```

Comment:
Use `errors.New` only when creating a contract `ErrXXX` API value. Since these are not saved as contract API variables, just return using `fmt.Errorf(...)`

Reply:
@lpabon okay